### PR TITLE
JWT token signed with RS256

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because it introduces an asymmetric algorithm for signing a JWT token.  The old token signed with HS256 will still work, but the new tokens will be signed with RS256. Adding RS256 will allow us to better handle token validation on App side.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/core/app.py [hunk 1]
```diff
@@ -13,3 +14,19 @@ def ready(self):
 
         if settings.SENTRY_DSN:
             settings.SENTRY_INIT(settings.SENTRY_DSN, settings.SENTRY_OPTS)
+
+    def validate_jwt_manager(self):
```
Comment (NyanKiyoshi) [inline saleor/core/app.py:18]: It doesn't seem like this method is unused anywhere

**Review by NyanKiyoshi — APPROVED** [review]
LGTM

File: saleor/core/jwt_manager.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                msg = (
+                    "RSA_PRIVATE_KEY is missing. Using temporary key for local "
+                    "development with DEBUG mode."
+                )
+                logger.warning(color_style().WARNING(msg))
+
+        try:
+            cls.get_private_key()
+        except Exception as e:
+            raise ImproperlyConfigured(f"Unable to load provided pem private key. {e}")
```
Comment (maarcingebala) [inline saleor/core/jwt_manager.py:155]: I think it should be referred to as PEM in the text:
```suggestion
            raise ImproperlyConfigured(f"Unable to load provided PEM private key. {e}")
```

File: saleor/core/jwt_manager.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        password = settings.RSA_PRIVATE_PASSWORD
+        if isinstance(password, str):
+            password = password.encode("utf-8")  # type: ignore
+        return serialization.load_pem_private_key(
+            pem, password=password  # type: ignore
+        )
+
+    @classmethod
+    def _load_debug_private_key(cls):
+        key_path = f"{settings.PROJECT_ROOT}/{cls.KEY_FILE_FOR_DEBUG}"
```
Comment (maarcingebala) [inline saleor/core/jwt_manager.py:71]: Would use `os.path.join` for that.

**Comment by patrys** [comment]
Task linked: [SALEOR-5833 Sign JWT tokens with RS256 instead of HS256](https://app.clickup.com/t/2549495/SALEOR-5833)

**Review by eCzerniak — APPROVED** [review]

**Pull request merged**